frontend/public/components/hpa.tsx

```diff
@@ -32,31 +32,186 @@ const menuActions = [
 ];
 
 export const HorizontalPodAutoscalersDetails: React.SFC<HorizontalPodAutoscalersDetailsProps> = ({obj: hpa}) => {
-  return <div className="co-m-pane__body">
-    <div className="row">
-      <div className="col-sm-6">
-        <ResourceSummary resource={hpa} showPodSelector={false} showNodeSelector={false} />
+  let type, current, target, currentMetrics;
+  const namespace = hpa.metadata.namespace;
+
+  // const mockCurrentMetrics = [
+  //   {
+  //     "type": "Object",
+  //     "object": {
+  //       "currentValue": "1k"
+  //     }
+  //   },
+  //   {
+  //     "type": "Pods",
+  //     "pods": {
+  //       "currentAverageValue": "100"
+  //     }
+  //   },
+  //   {
+  //     "type": "Resource",
+  //     "resource": {
+  //       "currentAverageUtilization": 5,
+  //       "currentAverageValue": "5",
+  //     }
+  //   },
+  //   {
+  //     "type": "Resource",
+  //     "resource": {
+  //       "currentAverageValue": "5",
+  //     }
+  //   }
+  // ];
+  // const mockCurrentMetrics = null;
+
+  // if (mockCurrentMetrics) {
```

Comment:
You might try reducing your `switch` statements down a bit, use `return` to your advantage in the `.map`:

```JavaScript
let currentMetrics = hpa.status.currentMetrics.map((currentMetric, i) => {
  switch(currentMetric.type) {
    case 'External':
      return [currentMetric.external.currentAverageValue, currentMetric.external.currentValue];
    case 'Object':
      return currentMetric.object.currentValue;
    case 'Pods':
      return currentMetric.pods.currentAverageValue;
    case 'Resource':
      return currentMetric.resource.name === 'cpu' ?
      [currentMetric.resource.currentAverageUtilization + '%', currentMetric.resource.currentAverageValue] :
      currentMetric.resource.currentAverageValue;
    default:
      return null;
  }
});
```
You can eliminate quite a few lines.

Reply:
Might even want to use a `Class` and break some of this out into separate functions...

Reply:
Heh, sorry but got me thinking, you can even eliminate the switch if you want to:

```JavaScript 

let currentMetrics = hpa.status.currentMetrics.map((currentMetric, i) => {
  return {
    'External' : [currentMetric.external.currentAverageValue, currentMetric.external.currentValue],
    'Object' : currentMetric.object.currentValue,
    'Pods' : currentMetric.pods.currentAverageValue,
    'Resource' : currentMetric.resource.name === 'cpu' ?
      [currentMetric.resource.currentAverageUtilization + '%', currentMetric.resource.currentAverageValue] :
      currentMetric.resource.currentAverageValue
  // the type will match a key above
  }[currentMetric.type] || null;
});
```
fewer lines + pretty clear key-value match.